src/potions.ts

```diff
@@ -434,3 +430,121 @@ export function bathroomFinance(embezzlers: number): void {
     }
   }
 }
+
+class VariableMeatPotions {
+  potion: Item;
+  effect: Effect;
+  duration: number;
+  softcap: number; // Number of turns to cap out variable bonus
+  meatBonusPerTurn: number; // meat% bonus per turn
+  cappedMeatBonus: number;
+
+  constructor(
+    potion: Item,
+    softcap: number,
+    meatBonusPerTurn: number,
+    duration?: number,
+    effect?: Effect
+  ) {
+    this.potion = potion;
+    this.effect = effect ?? effectModifier(potion, "Effect");
+    this.duration = duration ?? numericModifier(potion, "Effect Duration");
+    this.softcap = softcap;
+    this.meatBonusPerTurn = meatBonusPerTurn;
+    this.cappedMeatBonus = softcap * meatBonusPerTurn;
+  }
+
+  use(quantity: number): boolean {
+    return use(quantity, this.potion);
+  }
+
+  price(historical: boolean): number {
+    // If asked for historical, and age < 14 days, use historical.
+    // If potion is not tradeable, use retrievePrice instead
+    return this.potion.tradeable
+      ? historical && historicalAge(this.potion) < 14
+        ? historicalPrice(this.potion)
+        : mallPrice(this.potion)
+      : retrievePrice(this.potion);
+  }
+}
+
+function triangleNumber(n2: number, n1?: number) {
+  return 0.5 * (n2 * (n2 + 1) - (n1 ?? 0) * ((n1 ?? 0) + 1));
+}
+
+export function considerVariableMeatPotions(yachtzees: number, embezzlers: number): void {
+  const potions = [
+    new VariableMeatPotions($item`love song of sugary cuteness`, 20, 2),
+    new VariableMeatPotions($item`pulled yellow taffy`, 50, 2),
+    // new VariableMeatPotions($item`porcelain candy dish`, 500, 1),
+  ];
+
+  potions.forEach((potion) => {
+    const barfTurns = Math.max(0, estimatedTurns() - yachtzees - embezzlers);
+    const yachtzeeValue = 2000;
+    const embezzlerValue = baseMeat + 750;
+    const barfValue = (baseMeat * turnsToNC) / 30;
+
+    function valueNPotions(n: number): number {
+      const totalCosts = n * potion.price(false);
+      const totalDuration = n * potion.duration;
+      let cappedDuration = Math.max(0, totalDuration - potion.softcap + 1);
+      let decayDuration = Math.min(totalDuration, potion.softcap - 1);
+      let totalValue = 0;
+      [
+        [yachtzees, yachtzeeValue],
+        [embezzlers, embezzlerValue],
+        [barfTurns, barfValue],
+      ].forEach((arr) => {
```

Comment:
why won't this let me write a suggestion I have so many suggestions

```
const turnTypes = [[yachtzees, yachtzeeValue], etc];

for (const [quantity, value] of turnTypes) {
```
(the TL;DR is name that variable, use a for loop, and destructure the variable instead of taking it as an array